client/state/domains/management/reducer.js

```diff
@@ -72,22 +78,18 @@ export const saving = createReducer( false, {
  * @return {Object}        Updated state
  */
 export const items = createReducer( {}, {
-	[ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_RECEIVE ]: ( state, { cacheData } ) => ( { ...state, _contactDetailsCache: cacheData } ),
-	[ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_UPDATE ]:
-		( state, { cacheData } ) =>
-			merge( {}, state, { _contactDetailsCache: cacheData } ),
+	[ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_RECEIVE ]: ( state, { data } ) => ( { ...state, _contactDetailsCache: data } ),
+	[ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_UPDATE ]: ( state, { data } ) => {
+		return merge( {}, state, { _contactDetailsCache: data } );
+	},
 	[ DOMAIN_MANAGEMENT_WHOIS_RECEIVE ]: ( state, { domain, whoisData } ) => ( { ...state, [ domain ]: whoisData } ),
-	[ DOMAIN_MANAGEMENT_WHOIS_UPDATE ]: ( state, { domain, whoisData } ) => ( {
-		...state,
-		[ domain ]: {
-			...state[ domain ],
-			...whoisData
-		}
-	} )
+	[ DOMAIN_MANAGEMENT_WHOIS_UPDATE ]: ( state, { domain, whoisData } ) => {
+		return merge( {}, state, { [ domain ]: { ...state[ domain ], ...whoisData } } );
```

Comment:
Note that:

1. This expression above would be the way to do it with `assign` (whether from `Object`, `lodash`, or with ES6+ spread syntax)
2. `merge`, being recursive, allows the following:

```diff
-return merge( {}, state, { [ domain ]: { ...state[ domain ], ...whoisData } } );
+return merge( {}, state, { [ domain ]: whoisData } );
```

3. I still don't think we should use `merge`, which is too implicitly powerful and side-effectful to manipulate our state tree. This should be another place where we should use `keyedReducer`, but that requires the decoupling of domain-agnostic and domain-specific concerns within the `items` reducer [as I recommended](https://github.com/Automattic/wp-calypso/pull/14701#discussion_r121697611) in the original PR.